mod/profile/start.php

```diff
@@ -159,6 +220,105 @@ function _profile_topbar_menu($hook, $type, $return, $params) {
 }
 
 /**
+ * Register menu items for the user hover menu
+ *
+ * @param \Elgg\Hook $hook hook
+ *
+ * @return array
+ *
+ * @access private
+ *
+ * @since 3.0
+ */
+function _profile_user_hover_menu(\Elgg\Hook $hook) {
+
+	$user = $hook->getEntityParam();
```

Comment:
with some reordering this can be cleaner

```php
if (!elgg_is_logged_in()) {
	return;
}

$user = $hook->getEntityParam();
if (!$user instanceof \ElggUser || !$user->canEdit()) {
     return;
}
```